lib/sdoblockio.c

```diff
@@ -45,7 +42,25 @@ bool sdo_block_alloc(sdo_block_t *sdob)
 	}
 
 	if (memset_s(sdob->block, sdob->block_size, 0) != 0) {
-		LOG(LOG_ERROR, "SDOW memset() failed!\n");
+		LOG(LOG_ERROR, "SDOBlock memset() failed!\n");
+		return false;
+	}
+	return true;
+}
+
+bool sdo_block_alloc_with_size(sdo_block_t *sdob, size_t block_sz)
```

Comment:
1. How is this different from sdo_block_alloc()?

2. Can sdo_block_alloc() use sdo_block_alloc_with_size() internally, without having to write the internal logic twice?

3. Need to document the functions with respect to what they do, how are the arguments interpreted and what are the return value. Better to do this documentation at the time of the writing code itself.

Reply:
Updated